[recoveryservicesbackup] minor fix for long running operation customer issue

hiaga opened the pull request:
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

<a href="?expand=1&template=sdk_configuration_template.md">Click here</a> to open a PR for only SDK configuration.

hiaga committed:
Added long running operation for create protected item
Added headers in 202 response

pcsin commented:
Some of the automated checks are failing, and from the initial review the change is not following the async provisioning guide - https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#asynchronous-operations

Pls ensure the swagger lint checks pass.

josefree commented:
A PR converting this service to TypeSpec has been prepped by the Azure SDK team as part of the drive to move all existing Azure services to use TypeSpec.  The PR for this is
https://github.com/Azure/azure-rest-api-specs/pull/35709
.  The team owning this service will need to validate the PR, address any highlighted linter failures, make any service-specific changes to the PR, have a review with ARM, and complete the PR before any additional work can be done on the service.  Information regarding how to do this finalization and who to contact for assistance are at https://tspwebsitepr.z22.web.core.windows.net/typespec-azure/prs/2700/docs/migrate-swagger/01-get-started/#review-and-adjust-the-typespec.  Going forward, any change to the service spec will need to be performed with TypeSpec.

krdhruva commented:
Some of the automated checks are failing, and from the initial review the change is not following the async provisioning guide - https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#asynchronous-operations

Pls ensure the swagger lint checks pass.

hiaga committed:
update tsp file

hiaga committed:
compile and generate from tsp

hiaga committed:
updated bms json

ChenxiJiang333 committed:
Add suppressions for breaking changes in RecoveryServicesBackup

Added suppressions for breaking changes in the azure-mgmt-recoveryservicesbackup package, including deleted or renamed clients, models, and method parameters.

hiaga committed:
Added Lintdiff suppression for ProvisioningStateSpecifiedForLROPut

hiaga committed:
minor change

hiaga committed:
Merge branch 'main' into hiaga/docFixes
